File: leetcode/typescript/two-sum.ts

/* Leetcode problem 1: Two Sum. */

function twoSum(nums: number[], target: number): number[] {
  const table = new Map()
  for (let i = 0; i < nums.length; ++i) {
    const value = nums[i]
    const need = target - value
    if (table.has(need))
      return [i, table.get(need)]
    table.set(nums[i], i)
  }
  return []
};

File: leetcode/java/TwoSum.java

/* Leetcode problem 1: Two Sum. */

import java.util.HashMap;

class TwoSum
{
  public int[]
  twoSum (int[] nums, int target)
  {
    HashMap<Integer, Integer> table = new HashMap<> ();

    for (int i = 0; i < nums.length; ++i)
      {
        int value = nums[i];
        int need = target - value;
        if (table.containsKey (need))
          return new int[] { table.get (need), i };
        table.put (value, i);
      }

    return null;
  }
}
